production/helm/loki/templates/admin-api/deployment-admin-api.yaml

```
@@ -24,7 +24,7 @@ spec:
     metadata:
       labels:
         {{- include "enterprise-logs.adminApiLabels" . | nindent 8 }}
-        {{- with merge .Values.loki.podLabels .Values.adminApi.labels }}
+        {{- with merge (deepCopy .Values.loki.podLabels) .Values.adminApi.labels }}
```

Just to provide an alternative (then use the one you prefer), you could use instead:

```suggestion
        {{- with merge (dict) .Values.loki.podLabels .Values.adminApi.labels }}
```

I recommend only because the code is (from my subjective perspective) more clean and it's (arguably) faster without making a full copy and, instead, just merging the top-level keys.

---

Nice! That seemed to work and, I agree, cleaner! (especially for label and annotation objects)!

Updated in https://github.com/grafana/loki/pull/16700/commits/d6e6fd61a66e7d172e5b01f55f4e0aee94831513